server/src/main/scala/org/apache/livy/server/SessionServlet.scala

```
@@ -131,13 +131,18 @@ abstract class SessionServlet[S <: Session, R <: RecoveryMetadata](
       if (tooManySessions) {
         BadRequest(ResponseMessage("Rejected, too many sessions are being created!"))
       } else {
-        val session = sessionManager.register(createSession(request))
-        // Because it may take some time to establish the session, update the last activity
-        // time before returning the session info to the client.
-        session.recordActivity()
-        Created(clientSessionView(session, request),
-          headers = Map("Location" ->
-            (getRequestPathInfo(request) + url(getSession, "id" -> session.id.toString))))
+        val input = createSession(request)
+        if(input == null) {
+          BadRequest(ResponseMessage("Rejected, invalid value for ttl field!"))
```

should we throw an exception in `createSession()` and use appropriate message from that to return the `ResponseMessage` here? There could potentially be other reasons on why `createSession()` could return null, right?

---

I have updated the code to avoid returning null.